File: C_Sharp/C_Sharp/domain/validators/ReservationValidator.cs
namespace C_Sharp.domain
{
    public class ReservationValidator : IValidator<Reservation>
    {
        public void validate(Reservation entity)
        {
            string errors = "";
            if(entity.ExcursionId <= 0){
                errors += "ID must be greater than 0!\n";
            }
            if(entity.Name == ""){
                errors += "Name is null!\n";
            }
            if(entity.TicketsNumber<=0) {
                errors += "Number of tickets must be greater than 0!\n";
            }
            if(entity.PhoneNumber.ToString().Length != 9){
                errors += "Phone number must contain 9 digits!\n";
            }
            if(errors != ""){
                throw new ValidationException(errors);
            }
        }
    }
}

File: MPP-Java/Services/src/main/java/domain/validators/ReservationValidator.java
package domain.validators;

import domain.Reservation;

public class ReservationValidator implements Validator<Reservation> {
    @Override
    public void validate(Reservation entity) throws ValidationException {
        String errors = "";
        if(entity.getExcursionID() <= 0){
            errors += "ID must be greater than 0!\n";
        }
        if(entity.getName().equals("")){
            errors += "Name is null!\n";
        }
        if(entity.getTicketsNumber()<=0) {
            errors += "Number of tickets must be greater than 0!\n";
        }
        if(entity.getPhoneNumber().toString().length() != 9){
            errors += "Phone number must contain 9 digits!\n";
        }
        if(!errors.equals("")){
            throw new ValidationException(errors);
        }
    }
}
